packages/flutter_tools/lib/src/build_system/targets/web.dart

```diff
@@ -130,6 +131,22 @@ class Dart2JSTarget extends Target {
       ? PackageMap.globalGeneratedPackagesPath
       : PackageMap.globalPackagesPath;
     final File outputFile = environment.buildDir.childFile('main.dart.js');
+
+    // Dart defines are encoded as a JSON array, and parsed as List<Object>.
+    List<Object> dartDefines = const <Object>[];
+    if (environment.defines.containsKey(kDartDefines)) {
+      final String dartDefinesJson = environment.defines[kDartDefines];
+      try {
+        dartDefines = jsonDecode(dartDefinesJson);
+      } on FormatException catch (_) {
+        throw Exception(
```

Comment:
NICE

Reply:
Does the build system define a more specific exception type that this could be translated to instead?

Reply:
Since this is only used in a single place, I would leave it as is for now.